development/git_lfs.rst

```diff
@@ -0,0 +1,156 @@
+If you downloaded the binary release, install ``git-lfs`` by running the provided :file:`install.sh`.
+
+Most package managers also provide the ``git-lfs`` client.
+Since, LFS is a rapidly evolving technology, package managers will help you keep up with new ``git-lfs`` releases.
+For example, Mac users with Homebrew_ can simply run ``brew install git-lfs`` and ``brew upgrade git-lfs``.
+
+Once ``git-lfs`` is installed, run
+
+.. code-block:: bash
+
+   git config --global lfs.batch false
+   git lfs install
+
+to install the necessary hooks in your :file:`~/.gitconfig` file.
+
+.. _git-lfs-auth:
+
+Setting up Git LFS authentication
+=================================
+
+Credentials *are not needed to clone or pull* from LFS-backed repositories, but are needed for push operations.
+Only GitHub users in the LSST GitHub organization can authenticate with DM's storage service.
+
+Git LFS mandates the HTTPS transport protocol.
+Since many HTTPS authentications happening during a single clone, push or fetch operation, it is essential that you use a Git credential helper to work effectively with Git LFS.
+:ref:`We describe how to setup a credential helper for you system in the deverloper workflow documentation <git-credential-helper>`.
+
+Once a helper is setup, you can cache your credentials by cloning any of DM's LFS-backed repositories.
+
+.. code-block:: bash
+
+   git clone https://github.com/lsst/afwdata.git
+
+``git clone`` will ask you to authenticate with DM's git-lfs server::
+
+   Username for 'https://git-lfs.lsst.codes': <GitHub username OR blank>
+   Password for 'https://<git>@git-lfs.lsst.codes': <GitHub password, token OR blank>
+
+- If you are a member of the LSST GitHub organization, then you can use your GitHub username and password.
+- If you *also* have `GitHub's two-factor authentication <https://help.github.com/articles/about-two-factor-authentication/>`_ enabled, use a personal access token instead of a password. You can setup a personal token at https://github.com/settings/tokens.
+- If you are only interested in cloning or pulling, the 'Username' and 'Password' can be blank.
+
+Finally, the ``git clone`` will ask you to authenticate to ``s3.lsst.codes``::
+
+   Username for 'https://s3.lsst.codes': <Empty>
+   Password for 'https://s3.lsst.codes': <Empty>
+
+There is no username or password for LSST's S3 service.
+
+Once your credentials are cached, you won't need to repeat this process on your system (:ref:`unless you opted for the cache-based credential helper <>`).
+
+.. _git-lfs-using:
+
+Using Git LFS-enabled repositories
+==================================
+
+Git LFS operates transparently to the user.
+*Just use the repo as you normally would any other Git repo.*
+All of the regular Git commands just work, whether you are working with LFS-managed files or not.
+
+There are two caveats for working with LFS: use of HTTPS is required, and Git LFS must be told to track new binary file types.
```

Comment:
Users can continue to use git+git and git+ssh. `git-lfs` (the git extension) uses HTTPS. Because of this you will have to authenticate with git-lfs over HTTPS. But git still works independently with it's own protocols.